cas-server-core-services/src/main/java/org/apereo/cas/services/AbstractResourceBasedServiceRegistryDao.java

```diff
@@ -0,0 +1,288 @@
+package org.apereo.cas.services;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apereo.cas.util.LockedOutputStream;
+import org.apereo.cas.util.ResourceUtils;
+import org.apereo.cas.util.StringSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.io.Resource;
+import org.springframework.util.Assert;
+
+import javax.annotation.PreDestroy;
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * This is {@link AbstractResourceBasedServiceRegistryDao}.
+ *
+ * @author Misagh Moayyed
+ * @since 5.0.0
+ */
+public abstract class AbstractResourceBasedServiceRegistryDao implements ServiceRegistryDao {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [This class has too many methods, consider refactoring it.](https://www.codacy.com/app/mmoayyed/cas/file/2643640391/issues/source?bid=3256115&fileBranchId=3322154#l35)